pkg/api/validation/validation.go

```diff
@@ -2571,7 +2571,9 @@ func validateServicePort(sp *api.ServicePort, requireName, isHeadlessService boo
 func ValidateServiceUpdate(service, oldService *api.Service) field.ErrorList {
 	allErrs := ValidateObjectMetaUpdate(&service.ObjectMeta, &oldService.ObjectMeta, field.NewPath("metadata"))
 
-	if api.IsServiceIPSet(oldService) {
+	// ClusterIP should be mutable only for empty value, to allow assigning ClusterIP
+	// for service changing type to one that requires ClusterIP
+	if oldService.Spec.ClusterIP != "" {
```

Comment:
What about the other direction?  We've been a little lax in the definition here.  Do we think it is valid to change from ClusterIP to ExternalName types?